nvptx-run.c

```diff
@@ -235,6 +279,27 @@ This program has absolutely no warranty.\n",
   r = cuCtxCreate (&ctx, 0, dev);
   fatal_unless_success (r, "cuCtxCreate failed");
 
+  size_t mem;
+  if (!stack_size || verbose)
+    {
+      r = cuDeviceTotalMem (&mem, dev);
+      fatal_unless_success (r, "could not get available memory");
+      report_val (stderr, "Total device memory", mem);
+    }
+
+  size_t free_mem;
+  size_t dummy;
+  if (verbose)
+    {
+      /* Set stack size limit to 0 to get more accurate free_mem.  */
+      r = cuCtxSetLimit(CU_LIMIT_STACK_SIZE, 0);
+      fatal_unless_success (r, "could not set stack limit");
+
+      r = cuMemGetInfo (&free_mem, &dummy);
```

Comment:
Actually, doesn't `dummy` here (when given a better name) make obsolete the earlier `cuDeviceTotalMem` call?

  - `cuMemGetInfo`: <https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g808f555540d0143a331cc42aa98835c0>
  - `cuDeviceTotalMem`: <https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1gc6a0d6551335a3780f9f3c967a0fde5d>

Or, is _total amount of memory available for allocation by the CUDA context_ vs. _total amount of memory available on the device_ intentional?